docs/source/contributing.rst

```diff
@@ -4,38 +4,63 @@ Contributing
+In there, you can edit code and build.
 Once you're done, make a pull request to the main repo and describe what you've done -- and we'll take it from there!
 
+
+Connecting with the garak team & community
+------------------------------------------
+
+If you're going to contribute, it's a really good idea to reach out, so you have a source of help nearby, and so that we can make sure your valuable coding time is spent efficiently as a contributor.
+There are a number of ways you can reach out to us:
+
+* GitHub discussions: `<https://github.com/leondz/garak/discussions>`
+* Twitter: `<https://twitter.com/garak_llm>`
+* Discord: `<https://discord.gg/uVch4puUCs>`
+
+ We'd love to help, and we're always interested to hear how you're using garak
+ .
+
 Developing your own plugins
 ---------------------------
 
+The recipe isn't outlandish:
+
 * Take a look at how other plugins do it
-* Inherit from one of the base classes, e.g. :class:`garak.probes.base.Probe`
-* Override as little as possible
-* You can test the new code in at least two ways:
-   * Start an interactive Python session
-      * Import the model, e.g. `import garak.probes.mymodule`
-      * Instantiate the plugin, e.g. `p = garak.probes.mymodule.MyProbe()`
-   * Run a scan with test plugins
-      * For probes, try a blank generator and always.Pass detector: `python3 -m garak -m test.Blank -p mymodule -d always.Pass`
-      * For detectors, try a blank generator and a blank probe: `python3 -m garak -m test.Blank -p test.Blank -d mymodule`
-      * For generators, try a blank probe and always.Pass detector: `python3 -m garak -m mymodule -p test.Blank -d always.Pass`
-   * Get `garak` to list all the plugins of the type you're writing, with `--list_probes`, `--list_detectors`, or `--list_generators`
-
-Tests
------
+* Start a new module inheriting from one of the base classes, e.g. :class:`garak.probes.base.Probe`
+* Override as little as possible.
+
+
+Guides to writing plugins
+-------------------------
+
+So you'd like to build a new garak plugin? Great! Here are our tutorials.
+
+* :doc:`Building a garak generator <contributing.generator>` -- step-by-step guide to building an interface for a real API-based model service
+
+
+
+Testing
+-------
+
+You can test your code in a few ways:
+
+* Start an interactive Python session
+   * Import the model, e.g. `import garak.probes.mymodule`
+   * Instantiate the plugin, e.g. `p = garak.probes.mymodule.MyProbe()`
+* Get `garak` to list all the plugins of the type you're writing, with `--list_probes`, `--list_detectors`, or `--list_generators`
+* Run a scan with test plugins
+   * For probes, try a blank generator and always.Pass detector: `python3 -m garak -m test.Blank -p mymodule -d always.Pass`
+   * For detectors, try a blank generator and a blank probe: `python3 -m garak -m test.Blank -p test.Blank -d mymodule`
+   * For generators, try a blank probe and always.Pass detector: `python3 -m garak -m mymodule -p test.Blank -d always.Pass`
```

Comment:
This was already here in another form os just noting for awareness. There was some conversation about possible shifting the `test` plugins to be only available in testing. If we want to have guidance in docs that refers to using them like this for examples they will need to stay exposed, although we may be able to exclude them from being listed by `--list_*`

Reply:
Right. I think that discussion was only about the `Single` generator used solely in unit testing - the test plugins help troubleshoot a garak setup, so I'd prefer to have them available generally.